### PR TITLE
Upgrade to v2.24.0 of the AWS Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**

### CODE REVIEW

File: sdk/python/pulumi_aws/rds/cluster_instance.py [hunk 1]
```diff
@@ -64,6 +64,7 @@ class ClusterInstance(pulumi.CustomResource):
     identifier: pulumi.Output[str]
     """
     The indentifier for the RDS instance, if omitted, this provider will assign a random, unique identifier.
+    >>>>>>> Update terminology for use in Pulumi documentation
```
Comment (pgavlin) [inline sdk/python/pulumi_aws/rds/cluster_instance.py]: Where are these coming from...?

Comment (stack72) [reply sdk/python/pulumi_aws/rds/cluster_instance.py]: And that, my friend is a dodgy merge conflict fix and why reviews are awesome - will push a fix when I get to the airport and search for any more occurances

File: sdk/nodejs/ram/principalAssociation.ts [hunk 2]
```diff
@@ -5,9 +5,17 @@ import * as pulumi from "@pulumi/pulumi";
...
 
 /**
- * Provides a Resource Access Manager (RAM) principal association.
+ * Provides a Resource Access Manager (RAM) principal association. Depending if [RAM Sharing with AWS Organizations is enabled](https://docs.aws.amazon.com/ram/latest/userguide/getting-started-sharing.html#getting-started-sharing-orgs), the RAM behavior with different principal types changes.
  * 
- * > *NOTE:* For an AWS Account ID principal, the target account must accept the RAM association invitation after resource creation.
+ * When RAM Sharing with AWS Organizations is enabled:
+ * 
+ * - For AWS Account ID, Organization, and Organizational Unit principals within the same AWS Organization, no resource share invitation is sent and resources become available automatically after creating the association.
+ * - For AWS Account ID principals outside the AWS Organization, a resource share invitation is sent and must be accepted before resources become available. See the [`aws.ram.ResourceShareAccepter` resource](https://www.terraform.io/docs/providers/aws/r/ram_resource_share_accepter.html) to accept these invitations.
```
Comment (pgavlin) [inline sdk/nodejs/ram/principalAssociation.ts]: @lukehoban are these URLs okay?

Comment (pgavlin) [reply sdk/nodejs/ram/principalAssociation.ts]: Feels like we should be mapping them somehow.

**Pull request merged**